langserver/completion.go

```diff
@@ -325,34 +329,33 @@ func (s *server) completeLabel(ctx context.Context, completions *[]protocol.Comp
 		return err
 	}
 
+	labelName := location.Node.(*promql.Item).Val
+	var prevMatched string
 OUTER:
-	for i, name := range allNames {
+	for _, match := range fuzzy.Find(labelName, allNames) {
```

Comment:
In case the current `labelName` is empty, it should match with all labels. Apparently fuzzy returns zero matches in this case.

Same for label values.

Reply:
in that case 
```
if labelName is empty{
    completion = append(completion,...name)
    return 
}
```
should fix that?

Reply:
I'm not sure what that snippet does.

Maybe it makes sense to have a wrapper around `fuzzy.Match()` with the same type that matches all input strings in case there the search term is empty.

Reply:
Like

```go
func Match(pattern string, items []string){
  
    if pattern == "" {
        var ret []fuzzy.Match
        strings.Sort(items)
        
        for index, item := range items{
              // Some deduplication logic
              ...

              append(ret, fuzzy.Match{Str: item, Score: len(items) - index})
              return ret
        }
    }
    return fuzzy.Match(...)
}
```